File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/Account/AccountMaskingSettings.cs
namespace Psi.Data.Models.ClientConfigurationModels.Account
{
    public class AccountMaskingSettings : SettingsBaseHelper
    {
        public AccountMaskingSettings(ISettingsBase settingsBase) : base(settingsBase)
        {

        }

        [SettingKey("Account.Masking.LengthToShow")]
        public int MaskingLengthToShow
        {
            get { return GetIntValue(); }
            set { SetValue(value); }
        }

        //TODO:  This property to should be converted to a list of String, instead of relying on consumers of this setting to parse the comma delimited string.
        [SettingKey("Account.Masking.HideWelcomeBarDetail")]
        public string HideWelcomeBarDetail
        {
            get { return GetValue(); }
            set { SetValue(value); }
        }

        [SettingKey("Account.Masking.AccountHistoryMaskingEnabled")]
        public bool AccountHistoryMaskingEnabled
        {
            get { return GetBoolValue() && !string.IsNullOrWhiteSpace(AccountHistoryMaskingRegexMatchingPatterns); }
            set { SetValue(value); }
        }

        [SettingKey("Account.Masking.AccountHistoryMaskingRegexMatchingPatterns")]
        public string AccountHistoryMaskingRegexMatchingPatterns
        {
            get { return GetValue() ?? string.Empty; }
            set { SetValue(value); }
        }

        [SettingKey("Account.Masking.CheckCopyMaskingEnabled")]
        public bool CheckCopyMaskingEnabled
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }

        /// <summary>
        /// If true, then show the unmasked Member Number when the user clicks on their Member Number in the PageMemberInfo.ascx member info control.
        /// </summary>
        [SettingKey("Account.Masking.ShowUnmaskedMemberNumberOnUserClick")]
        public bool ShowUnmaskedMemberNumberOnUserClick
        {
            get { return GetBoolValue(); }
            set { SetValue(value); }
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/Account/AccountMaskingSettings.ts
// Generated imports

export interface AccountMaskingSettings {
    /** @settingKey Account.Masking.LengthToShow */
    maskingLengthToShow: number;
    /** @settingKey Account.Masking.HideWelcomeBarDetail */
    /**
     * //TODO:  This property to should be converted to a list of String, instead of relying on consumers of this setting to parse the comma delimited string.
     */
    hideWelcomeBarDetail: string;
    /** @settingKey Account.Masking.AccountHistoryMaskingEnabled */
    accountHistoryMaskingEnabled: boolean;
    /** @settingKey Account.Masking.AccountHistoryMaskingRegexMatchingPatterns */
    accountHistoryMaskingRegexMatchingPatterns: string;
    /** @settingKey Account.Masking.CheckCopyMaskingEnabled */
    checkCopyMaskingEnabled: boolean;
    /** @settingKey Account.Masking.ShowUnmaskedMemberNumberOnUserClick */
    /**
     * /// <summary>
     * /// /// <summary>
     * /// /// If true, then show the unmasked Member Number when the user clicks on their Member Number in the PageMemberInfo.ascx member info control.
     * /// /// </summary>
     * /// </summary>
     */
    showUnmaskedMemberNumberOnUserClick: boolean;
}
